interrupterror.go

```diff
@@ -0,0 +1,13 @@
+package miniline
+
+// InterruptedError represents the user having exited the prompt with ^C
+type InterruptedError struct{}
+
+// Error is just the string "Interrupted"
+func (e InterruptedError) Error() string {
+	return "Interrupted"
+}
+
+// ErrInterrupted is a singleton InterruptedError
```

Comment:
Comments should not be totally obvious :)

Maybe make it // ErrInterrupted is an error if input is interrupted